vector/src/main/java/im/vector/util/DinsicUtils.java

```diff
@@ -34,4 +50,72 @@ public  static boolean isFromFrenchGov(final List<String> emails) {
         return myReturn;
     }
 
+    /**
+     * Edit contact form
+     */
+    public static void editContactForm(Context theContext, Activity myActivity, String editContactWarningMsg, final Contact contact ) {
+        LayoutInflater inflater = LayoutInflater.from(theContext);
+
+        Cursor namesCur=null;
+        boolean switchToContact=false;
+        try
+        {
+            ContentResolver cr = theContext.getContentResolver();
+            namesCur = cr.query(ContactsContract.Data.CONTENT_URI,
+                    new String[]{ContactsContract.Contacts.DISPLAY_NAME_PRIMARY,
+                            ContactsContract.Contacts.LOOKUP_KEY,
+                            ContactsContract.CommonDataKinds.StructuredName.CONTACT_ID
+                    },
+                    ContactsContract.Data.MIMETYPE + " = ? AND "+ ContactsContract.Data.CONTACT_ID + " = ?",
+
+                    new String[]{ContactsContract.CommonDataKinds.StructuredName.CONTENT_ITEM_TYPE,
+                            contact.getContactId()}, null);
+            if (namesCur != null) {
+                if (namesCur.moveToNext()) {
+                    String contactId = namesCur.getString(namesCur.getColumnIndex(ContactsContract.CommonDataKinds.StructuredName.CONTACT_ID));
+                    Uri mSelectedContactUri;
+                    int mLookupKeyIndex;
+                    int mIdIndex;
+                    String mCurrentLookupKey;
+                    mLookupKeyIndex = namesCur.getColumnIndex(ContactsContract.Contacts.LOOKUP_KEY);
+                    mCurrentLookupKey = namesCur.getString(mLookupKeyIndex);
+                    //mCursor.getColumnIndex(ContactsContract.Contacts._ID);
+                    long mCurrentId = Integer.parseInt(contactId);//namesCur.getLong(mIdIndex);
+                    mSelectedContactUri =
+                            ContactsContract.Contacts.getLookupUri(mCurrentId, mCurrentLookupKey);
+
+
+                    Intent editIntent = new Intent(Intent.ACTION_EDIT);
+                    editIntent.setDataAndType(mSelectedContactUri, ContactsContract.Contacts.CONTENT_ITEM_TYPE);
+                    editIntent.putExtra("finishActivityOnSaveCompleted", true);
+                    myActivity.startActivity(editIntent);
+                    switchToContact=true;
+                }
+            }
+        } catch (Exception e) {
+            Log.e(LOG_TAG, "## editContactForm(): Exception - Msg=" + e.getMessage());
+        }
+        if (!switchToContact){
+            AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(myActivity);
+            alertDialogBuilder.setMessage(editContactWarningMsg);
+
+            // set dialog message
+            alertDialogBuilder
+                    .setCancelable(false)
+                    .setPositiveButton(R.string.ok,
+                            new DialogInterface.OnClickListener() {
+                                public void onClick(DialogInterface dialog, int id) {
+
```

Comment:
Why no action is expected when you click OK ?

Reply:
modified